micro-rdk/src/esp32/board.rs

```diff
@@ -365,4 +380,44 @@ impl Board for EspBoard {
         }
         Err(BoardError::GpioPinError(pin as u32, "not configured"))
     }
+
+    fn add_digital_interrupt_callback(
+        &mut self,
+        pin: i32,
+        intr_type: InterruptType,
+        cb: gpio_isr_t,
+        arg: Option<*mut core::ffi::c_void>,
+    ) -> Result<(), BoardError> {
+        let p = self
+            .pins
+            .iter_mut()
+            .find(|p| p.pin() == pin)
+            .ok_or_else(|| {
+                BoardError::GpioPinError(
+                    pin as u32,
+                    "pin not found, failed to register interrupt callback",
+                )
+            })?;
+
+        if cb.is_some() {
+            p.setup_interrupt(intr_type, cb, arg.unwrap_or_else(core::ptr::null_mut))?;
+        } else {
+            let ptr = &mut p.event_count as *mut Arc<std::sync::atomic::AtomicU32> as *mut _;
```

Comment:
since the  `event_count` is a field in the individual `Esp32GPIOPin` and pins are instantiated once per session, this should be fine.

Reply:
what is the risk  otherwise?

Reply:
that the address of `event_count` might change if the `Esp32GPIOPin` moves somehow, or if the pin is re-instantiated, there might be a dangling pointer as the registered arg at the esp-idf level.